test/saithrift/tests/saitunnel.py

```diff
@@ -0,0 +1,321 @@
+        #  Egress configurations
+        #  create router interface,
+        #  create neighbor
+        #  create nhop
+        #  create route
+        #   
+        ##############################################################################
+       
+        #encap router interface
+        encap_if_id=sai_thrift_create_router_interface(self.client, vr_id=vr_id, is_port=1, port_id=port1, vlan_id=vlan_id, v4_enabled = 1, v6_enabled = 1, mac='', is_lb=0)
+        #egress(encap side) neighbor (ip=10.10.1.1 , mac=00:05:06:00:00:00 )
+        sai_thrift_create_neighbor(self.client, addr_family=addr_family, rif_id=encap_if_id, ip_addr=encap_ip_addr, dmac=neighbor_mac_encap)
+        #egress(encap) nhop and route create
+        egress_nhop=sai_thrift_create_nhop(self.client, addr_family=addr_family, ip_addr=egress_nhop_ip_addr, rif_id=encap_if_id, is_tunnel=0)
+        sai_thrift_create_route(self.client, vr_id=vr_id, addr_family=addr_family, ip_addr=nhop_ip_addr_encap, ip_mask=nhop_ip_mask_encap, nhop=egress_nhop)
+        
+        ###############################################################################
+        #  Ingress configurations
+        #  create router interface,
+        #  create neighbor
+        #  create nhop
+        #  create route
+        #   
+        ##############################################################################
+         
+        #ingress router interface
+        ingress_if_id=sai_thrift_create_router_interface(self.client,vr_id=vr_id,is_port=1,port_id=port2,vlan_id=vlan_id,v4_enabled=1,v6_enabled = 1, mac='', is_lb=0)
+        #ingress neighbor neighbor (ip=20.20.1.1 , mac = 00:05:06:00:00:02)
+        sai_thrift_create_neighbor(self.client, addr_family=addr_family, rif_id=ingress_if_id, ip_addr=ingress_ip_addr, dmac=neighbor_mac_ingress)
+        #ingress nhop and route create
+        ingress_nhop=sai_thrift_create_nhop(self.client, addr_family=addr_family, ip_addr=ingress_nhop_ip_addr, rif_id=ingress_if_id, is_tunnel=0)
+        sai_thrift_create_route(self.client, vr_id=vr_id, addr_family=addr_family, ip_addr=nhop_ip_addr_ingress, ip_mask=nhop_ip_mask_ingress, nhop=ingress_nhop)
+
+        
+        #adding tunnel and route
+        initiator_id=sai_thrift_create_nhop(self.client, addr_family=addr_family, ip_addr=initiator_ip_addr, rif_id=tunnel_id , is_tunnel=1)
+        sai_thrift_create_route(self.client,vr_id=vr_id, addr_family=addr_family, ip_addr=tunnel_ip_addr, ip_mask=tunnel_ip_mask, nhop=initiator_id)
+        
+        #Packet to be send 
+        pkt = simple_tcp_packet(eth_dst=router_mac,
+                                eth_src='00:00:00:00:00:01',
+                                ip_src='20.20.1.2',
+                                ip_dst='1.1.1.1',
+                                ip_id=1,
+                                ip_ttl=64)
+        #expected packet inner IP header
+        inner_hdr = simple_tcp_packet(eth_dst=router_mac,

```

Comment:
We are taking only IP header while passing arguments to inner_frame in simple_ipv4ip_packet

inner_hdr = simple_tcp_packet(eth_dst=router_mac,
                                eth_src='00:00:00:00:00:01',
                                ip_src='20.20.1.2',
                                ip_dst='1.1.1.1',
                                ip_id=1,
                                ip_ttl=63)
        exp_pkt = simple_ipv4ip_packet(eth_dst='00:05:06:00:00:00',
                                eth_src=router_mac,
                                ip_dst='10.10.1.1',
                                ip_src='2.2.2.2',
                                ip_id=0,
                                ip_ttl=64,
                                inner_frame=inner_hdr['IP']
                                )